### ISSUE TITLE
limo: certain icons are missing

### ISSUE THREAD

**Issue by poperigby** [NixOS/nixpkgs#365287]
## Describe the bug

Since #342877 was merged, I thought I would open an issue to get this fixed.

The issue is that quite of few of the icons are missing, as shown here: https://github.com/NixOS/nixpkgs/pull/342877#issuecomment-2397133222

See this thread for more details: https://github.com/NixOS/nixpkgs/pull/342877#pullrequestreview-2361713198

## Expected behavior

All icons show up correctly.

## Screenshots

![Screenshot_2024-10-07_07-39](https://github.com/user-attachments/assets/4375caa8-d5ac-4397-8e90-1462d5409c40)

## Metadata

```
- system: `"x86_64-linux"`
 - host os: `Linux 6.12.3, NixOS, 25.05 (Warbler), 25.05.20241211.5d67ea6`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/cassidy/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/cassidy/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/cassidy/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/cassidy/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/smghprxmwyp707kqljlm4fsxm7h5wd72-lix-2.91.1/share`
 - nixpkgs: `/nix/store/rhn4fkh7hqjlqs9j6naaq623qpl62yz0-source`
```

## Notify maintainers

@TomaSajt @MattSturgeon 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by chewblacka** [comment]
Can someone test whether the PR fixes this issue? Many thanks.

**Comment by poperigby** [comment]
It seems hicolor doesn't provide these icons, but I was able to fix it by setting my `QT_QPA_PLATFORMTHEME` to `gtk2`. I'm on Sway.

**Comment by MattSturgeon** [comment]
> I was able to fix it by setting my `QT_QPA_PLATFORMTHEME` to `gtk2` or `gtk3`. I'm on Sway.

Interesting. We could set that in the wrapper script. However I'm unsure if we'd want the value to be conditional, maybe DE-specific? :thinking:

**Comment by poperigby** [comment]
Really, the DEs should be setting that environment variable. Has someone tested this on GNOME or KDE?

**Comment by MattSturgeon** [comment]
> Has someone tested this on GNOME or KDE?

All my testing has been on GNOME... :slightly_frowning_face:

**Comment by poperigby** [comment]
Did you check the value of that variable? Do any other Qt apps have problems with missing icons? There's nothing special about Limo.

**Comment by MattSturgeon** [comment]
> Did you check the value of that variable?

It's unset on my system.

Setting it to `gtk2` or `gtk3` seems to make no difference either...
```
$ NIXPKGS_ALLOW_UNFREE=1 nix run nixpkgs#limo --impure
error: unable to execute '/nix/store/l2j2jqp86ih3i7270bsvqpdawsagb0ls-limo-1.0.10/bin/Limo': No such file or directory

$ NIXPKGS_ALLOW_UNFREE=1 nix shell nixpkgs#limo --impure

$ QT_QPA_PLATFORMTHEM=gtk2 limo
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.

$ QT_QPA_PLATFORMTHEM=gtk3 limo
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
```

Note that _one or two_ icons do show, but those ones showed up before anyway. They look like hicolor icons to my eye, but I could be wrong.

(also just discovered `meta.mainProgram` has a typo, so `nix run` doesn't work :sweat_smile:

**Comment by keenanweaver** [comment]
This is what I see on KDE without any environment variables. I don't have any 'supported' games installed right now to test those.

![image](https://github.com/user-attachments/assets/d2008e07-88a5-40ac-9144-a941eb48a4f6)

**Comment by MattSturgeon** [comment]
> This is what I see on KDE without any environment variables.

Looks like icons are working on your system. For example those "edit" buttons are blank for me.

Do you know what icon theme those icons are coming from? And/or what icon themes you have available?

**Comment by poperigby** [comment]
Any icon theme that has the required icons should work, but they look like Papirus to me. I'm thinking this is an environment specific issue unrelated to Limo, because Limo is simply pulling from the system icon theme. @MattSturgeon that environment variable definitely shouldn't be unset normally, unless you're using a do-it-yourself WM like Sway.

**Comment by keenanweaver** [comment]
> Do you know what icon theme those icons are coming from? And/or what icon themes you have available?

My installed icon themes are: 
`papirus-icon-theme` (primary)
`material-design-icons`
`hicolor-icon-theme`
`adwaita-icon-theme`

**Comment by MattSturgeon** [comment]
Thanks for the info.

Playing around a little I seems I _don't_ need `QT_QPA_PLATFORMTHEM` set, but I do need to change my default icon theme in gnome-tweaks or home-manager.

- When `hicolor` or `adwiter` are selected, many icons are missing.
- When `breeze` or `papirus` are enabled, all icons seem to show up.

So I guess this means two things:
1. icon fallthrough isn't working
2. some required icons are not present in hicolor/adwiter

I'm not really sure what to _do_ with that information, but hopefully it helps someone to debug this issue further :crossed_fingers:

**Comment by poperigby** [comment]
> that environment variable definitely shouldn't be unset normally, unless you're using a do-it-yourself WM like Sway

Yeah I'm actually probably wrong on that. The intricacies of Linux desktop theming will drive one insane.
